app/code/Magento/CatalogWidget/view/frontend/templates/product/widget/content/grid.phtml

```diff
@@ -74,6 +74,12 @@ use Magento\Wishlist\Helper\Data;
                                                     <?php if ($_item->isSaleable()): ?>
                                                         <?php $postParams = $block->getAddToCartPostParams($_item); ?>
                                                         <form data-role="tocart-form" data-product-sku="<?= $escaper->escapeHtml($_item->getSku()) ?>" action="<?= $escaper->escapeUrl($postParams['action']) ?>" method="post">
+                                                            <?php $options = $block->getViewModel()->getOptionsData($_item); ?>
```

Comment:
Will be complete solution if you add the same for the pagebuilder package https://github.com/magento/magento2-page-builder/blob/4a2795ea9d2dae7cec8437e5e25e78e67c1b8d90/app/code/Magento/PageBuilder/view/frontend/templates/catalog/product/widget/content/carousel.phtml#L62

Reply:
Same as here: https://github.com/magento/magento2-page-builder/pull/868/files#r1560503440
`<?php $options = $block->getViewModel()->getOptionsData($_item); ?>`
change it to 
`<?php if($options = $block->getViewModel()?->getOptionsData($_item)) : ?>`
will prevent empty/not assigned viewModel

and ofc, close if, at the end of foreach.

Reply:
@Bashev Thanks for your suggestion. I've updated the code as per your suggestion